src/lib/hooks/transactions/updater.tsx

```diff
@@ -49,15 +52,27 @@ export default function Updater({ pendingTransactions, onCheck, onReceipt }: Upd
 
   const lastBlockNumber = useBlockNumber()
   const fastForwardBlockNumber = useFastForwardBlockNumber()
+  const blockTimestamp = useCurrentBlockTimestamp()
 
   const getReceipt = useCallback(
     (hash: string) => {
       if (!provider || !chainId) throw new Error('No provider or chainId')
       const retryOptions = RETRY_OPTIONS_BY_CHAIN_ID[chainId] ?? DEFAULT_RETRY_OPTIONS
+      const deadline = (pendingTransactions[hash].info as ExactOutputSwapTransactionInfo).deadline
       return retry(
         () =>
           provider.getTransactionReceipt(hash).then((receipt) => {
-            if (receipt === null) {
+            if (receipt === null && blockTimestamp) {
+              if (blockTimestamp.gt(deadline)) {
+                onReceipt({
+                  chainId,
+                  hash,
+                  receipt: {
+                    ...(receipt as TransactionReceipt),
```

Comment:
this is effectively `...(null as TransactionReceipt)`, right?

I think a better approach would be updating `getReceipt` to explicitly remove the transaction from the app transactions store instead of forcing a faked transaction to "confirm"